product_unavailable/models/product_template.py

```diff
@@ -11,3 +11,18 @@ class ProductTemplate(models.Model):
     availability = fields.Selection(
         selection_add=[('unavailable', 'Unavailable')],
     )
+    availability_sequence = fields.Integer(
+        compute='_compute_availability_sequence',
+        store=True,
+    )
+
+    @api.depends('availability')
+    def _compute_availability_sequence(self):
+        sequence_dict = {
```

Comment:
Move this to a class var so modules adding statuses can also add sorting without modifying the method